tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -59,12 +60,13 @@ syntaxDatatype language datatype = do
       con <- ctorForProductType datatypeName typeParameterName children fields
       result <- symbolMatchingInstance language name datatypeName
       pure $ generatedDatatype name [con] typeParameterName:result
-    LeafType (DatatypeName datatypeName) named -> do
-      con <- ctorForLeafType named (DatatypeName datatypeName) typeParameterName
+    LeafType (DatatypeName datatypeName) Anonymous -> do
+      tsSymbol <- runIO $ withCString datatypeName (TS.ts_language_symbol_for_name language)
+      pure [ TySynD name [] (ConT ''Token `AppT` LitT (StrTyLit datatypeName) `AppT` LitT (NumTyLit (fromIntegral tsSymbol))) ]
```

Comment:
I opted to split these cases up at the top level `case` because they have basically no overlap any more—we don’t have to build a constructor or instance declarations for anonymous leaves any more.